client/state/sites/plans/connections.js

```diff
@@ -0,0 +1,59 @@
+import debugFactory from 'debug';
+import map from 'lodash/map';
+
+import {
+	SITE_PLANS_FETCH_FAILED,
+	SITE_PLANS_TRIAL_CANCEL_COMPLETED,
+	SITE_PLANS_TRIAL_CANCEL_FAILED
+} from 'state/action-types';
+import { createSitePlanObject } from './assembler';
+import { fetchSitePlansCompleted } from './actions';
+import i18n from 'i18n-calypso';
+
+const debug = debugFactory( 'calypso:site-plans:connections' );
+
+export const connectCancelSitePlanTrial = wpcom => ( action, dispatch ) => {
+	return new Promise( ( resolve, reject ) => {
```

Comment:
It doesn't need to return Promise anymore, as we don't check against return value with this implementation :)